### PR TITLE
Fix Hyperlink stays in pressed state when releasing pointer out

### PR DESCRIPTION
**Pull request by dr1rrb**
## Bugfix
Pressed state of `Hyperlink`

## What is the current behavior?
When pressing over an `Hyperlink` (which is not in a `ScrollViewer`), moving pointer out it, and then release the pointer, the `Hyperlink` stays in pressed state.

## What is the new behavior?
Pressed state is released.

## PR Checklist
- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] ~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] ~Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)~
- [x] Associated with an issue (GitHub or internal)

## Other information
Fixes https://github.com/unoplatform/private/issues/41
Internal Issue: https://nventive.visualstudio.com/_workitems/edit/159324

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/VisualState_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+			Run("UITests.Shared.Windows_UI_Input.VisualStatesTests.Buttons");
+
+			var rect = _app.WaitForElement(target).Single().Rect;
+
+			// Press over and move out to release
+			_app.DragCoordinates(rect.X + 2, rect.Y + 2, rect.X, rect.Y - 30);
+
+			// Validate the state was restored to default
+			TakeScreenshot("Result");
```
Comment (MatFillion) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/VisualState_Tests.cs]: the test only takes a screenshot?

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs [hunk 2]
```diff
@@ -695,7 +695,12 @@ private void UpdateInlines(string text)
 				// KNOWN ISSUE:
 				// On UWP the 'click' event is raised **after** the PointerReleased ... but deferring the event on the Dispatcher
 				// would move it after the PointerExited. So prefer to raise it before (actually like a Button).
-				that.FindHyperlinkAt(e.GetCurrentPoint(that).Position)?.ReleasePointerPressed(e.Pointer);
+				if (!(that.FindHyperlinkAt(e.GetCurrentPoint(that).Position)?.ReleasePointerPressed(e.Pointer) ?? false))
+				{
+					// We failed to find the hyperlink that made this capture but we ** silently ** removed the capture,
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs]: Confirm my understanding, if the pointer is released outside an hyperlink, we won't "find" it, so we abort captures on all our hyperlinks?

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs]: A pointer up will release the capture made for it. The issue is that we muted the event `PointerCaptureLost` (to match UWP behavior), but we didn't find the hyperlink which made the capture, so here we only make sure to notify the hyperlink that had captured this pointer to go back in default state.

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/VisualState_Tests.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/VisualState_Tests.cs]: I agree, detecting state is required here, not just a screenshot, detecting changes numbers with a counter may be helpful there as well.

Comment (dr1rrb) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Input/VisualState_Tests.cs]: Yes I agree also, I've already done a micro diff tool for screenshots (not tested yet), and indeed while writing this reply I realized that I probably have a way to validate effective states changes: I only have to listen the `ViusalStateManager` 😃

**Comment by jeromelaban** [comment]
Can you enable those tests for iOS ?

https://github.com/unoplatform/uno/blob/7afd6adaa4a6a3cdd7bf75bd4d50e5d67b1a6e07/build/ios-uitest-run.sh#L30-L32

File: src/Uno.UI/UI/Xaml/UIElement.wasm.cs [hunk 3]
```diff
@@ -384,16 +380,13 @@ public bool Dispatch(EventArgs eventArgs, string nativeEventPayload)
 						routedArgs.CanBubbleNatively = _canBubbleNatively;
 					}
 
-					if (_eventFilterManaged(args))
+					foreach (var handler in _invocationList)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/UIElement.wasm.cs]: Would invocation list change during enumeration ?

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/UIElement.wasm.cs]: I didn't changed this code, but it's why there is a `_pendingInvocationList`: https://github.com/unoplatform/uno/pull/2054/files/940cff5808342a2ec7fa130c830bb914f6f184ad#diff-6f3c127d07fda497219e71a7d9b763a1R323

**Pull request merged**